geopyspark/geotrellis/tms.py

```diff
@@ -115,18 +116,86 @@ class TMS(object):
     Attributes:
         pysc (pyspark.SparkContext): The ``SparkContext`` being used this session.
         server (JavaObject): The Java TMSServer instance
+        host (str): The IP address of the host, if bound, else None
+        port (int): The port number of the TMS server, if bound, else None
+        url_pattern (string): The URI pattern for the current TMS service, with
+            {z}, {x}, {y} tokens.  Can be copied directly to services such as
+            `geojson.io`.
     """
 
     def __init__(self, server):
         self.pysc = get_spark_context()
         self.server = server
         self.handshake = ''
+        self.bound = False
+        self._host = None
+        self._port = None
         self.pysc._gateway.start_callback_server()
 
     def set_handshake(self, handshake):
         self.server.set_handshake(handshake)
         self.handshake = handshake
 
+    def bind(self, host, requested_port=None):
+        """Starts up a TMS server.
+
+        Args:
+            host (str): The target host.  Typically "0.0.0.0", "localhost", or
+                "127.0.0.1".  The first will make the TMS service accessible
+                from the world.
+
+            requested_port (optional, int): A port number to bind the service
+                to.  If omitted, a random port.
+        """
+        try:
+            if requested_port:
+                self.server.bind(host, requested_port)
+            else:
+                self.server.bind(host)
+            self.bound = True
+            self._port = self.server.port()
+            self._host = [l for l in
+                          ([ip for ip in socket.gethostbyname_ex(socket.gethostname())[2] if not ip.startswith("127.")][:1],
+                           [[(s.connect(('8.8.8.8', 53)), s.getsockname()[0], s.close()) for s in
+                             [socket.socket(socket.AF_INET, socket.SOCK_DGRAM)]][0][1]])
+                          if l][0][0]
```

Comment:
This behavior I think is the least confusing:
- `host` is unspecified => bind to `localhost`
   this is the safest choice
- `host` is `0.0.0.0` => perform reverse lookup
   0.0.0.0 implies that this this service is supposed to be accessible from outside, so lets do our best.
- `host` is a hostname => bind and report it as given
   Presumably they gave a hostname that somehow binds to an interface, so it should work on those terms.

AFAIK `0.0.0.0` is the only case where we break hostname, interface  1-1 mapping and have to do something special to get a useful URI.